File: script/mode.py
from collections import Counter

def mode(num):
	count_dict = dict(Counter(num).most_common())
	max_count = Counter(num).most_common(1)[0][1]
	modes = []
	for x in num:
		if count_dict[x] == max_count: 
			modes.append(x)
	mode = sorted(modes)[0]
	return(mode)

print mode([1087607, 1087628])
print mode([1220051,1220059,1220063,1220068])

File: script/mode.pl
#sub mode {
#	my @nums = @_;
#	my %ha;
#	foreach my $i (@nums){
#		$ha{$i} ++;
#	}
#	my ($mode) = (sort {$ha{$a} <=> $ha{$b}} keys %ha)[-1];
#	return $mode;
#}

sub mode {
	my @nums = @_;
	my %ha;
	foreach my $i (@nums){
		$ha{$i} ++;
	}
	my $max_count = (sort{$a<=>$b} values %ha)[-1];
	print "$max_count\n";
	my @modes;
	foreach my $n (@nums) {
		print "$n\n";
		if ($ha{$n} == $max_count) {
			push @modes, $n;		
		}
	}
	my $mode = (sort @modes)[0]; 
	return $mode;
}



my @num = (1220051,1220059,1220063,1220068,1220063,1220068);

my $m = mode(@num);

print "mode = $m\n";

